addons/mail/static/src/models/notification_group/notification_group.js

```diff
@@ -67,6 +67,28 @@ function factory(dependencies) {
             return `${this.modelName}_${data.id}`;
         }
 
+        _computeDate() {
+            const groupDate = moment.max(this.notifications.map(notification => notification.message.date));
```

Comment:
message is currently not set as required on notification, so either the required flag should be added, or this code should handle the case when it's not